pkg/nirmata/nirmata_ProviderManaged_cluster.go

```diff
@@ -101,7 +102,42 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error {
 }
 
 func resourceClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	apiClient := meta.(client.Client)
+	nodeCount := d.Get("node_count").(int)
+	name := d.Get("name").(string)
 
+	clusterTypeID, err := apiClient.QueryByName(client.ServiceClusters, "KubernetesCluster", name)
+	if err != nil {
+		fmt.Printf("Error ", err)
+		return err
+	}
+	data, err := apiClient.Get(clusterTypeID, &client.GetOptions{})
+	if err != nil {
+		fmt.Printf("Error ", err)
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	nodePools := data["nodePools"].([]interface{})
+	for _, nodePool := range nodePools {
```

Comment:
Instead of iterating over the node pools, the code could be improved if you check the size of the node pools, return and error if its < 1 and then extract and update the first node pool.

Currently, its possible that no node pools are returned and there is no error reported.